### PR TITLE
[SPARK-30418][ML] Make FM call super class method extractLabeledPoints

### PR DESCRIPTION
**Pull request by huaxingao**
### What changes were proposed in this pull request?
make FMClassifier/Regressor call super class method extractLabeledPoints

### Why are the changes needed?
code reuse

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
existing tests

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/ml/classification/FMClassifier.scala [hunk 1]
```diff
@@ -204,14 +204,8 @@ class FMClassifier @Since("3.0.0") (
...
     val handlePersistence = dataset.storageLevel == StorageLevel.NONE
-    val data: RDD[(Double, OldVector)] =
-      dataset.select(col($(labelCol)), col($(featuresCol))).rdd.map {
-        case Row(label: Double, features: Vector) =>
-          require(label == 0 || label == 1, s"FMClassifier was given" +
-            s" dataset with invalid label $label.  Labels must be in {0,1}; note that" +
-            s" FMClassifier currently only supports binary classification.")
-          (label, features)
-      }
+    val labeledPoint = extractLabeledPoints (dataset, numClasses)
```
Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/ml/classification/FMClassifier.scala]: Nit: remove extra space before method call

Comment (huaxingao) [reply mllib/src/main/scala/org/apache/spark/ml/classification/FMClassifier.scala]: Removed. Thanks!

**Comment by srowen** [comment]
Merged to master

**Comment by huaxingao** [comment]
Thank you Sean!

**Pull request closed without merging**